libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -135,6 +139,33 @@ public static String getSHA256Hash(String privateKey) {
         return hashedString;
     }
 
+    /**
+     * Generate a RSA-2048 keypair and get the encoded public key string
+     *
+     * @return RSA-2048 public key string
+     */
+    @Nullable
+    public static synchronized String getRSA2048PublicString(String name) {
+        final SharedPreferences prefs = SalesforceSDKManager.getInstance().getAppContext().getSharedPreferences(SHARED_PREF_FILE, 0);
```

Comment:
I'd prefer not to reuse the existing shared pref file that houses symmetric encryption keys for this. Let's create a new one.